thirtyinch-rx2/src/test/java/net/grandcentrix/thirtyinch/rx2/RxTiPresenterUtilsTest.java

```diff
@@ -57,4 +59,116 @@ public void testIsViewReady_DisposeBeforeAttachView_ShouldRemoveCallback() throw
         mPresenter.attachView(mView);
         test.assertValue(false);
     }
-}
\ No newline at end of file
+
+    @Test
+    public void testDeliverLatestToView_ViewNotReady() throws Exception {
+        mPresenter.create();
+
+        TestObserver<Integer> testObserver = new TestObserver<>();
+        Observable.just(1, 2, 3)
+                .compose(RxTiPresenterUtils.<Integer>deliverLatestToView(mPresenter))
+                .subscribe(testObserver);
+
+        mPresenter.attachView(mView);
+
+        testObserver.assertNotComplete();
+        testObserver.assertNoErrors();
+        testObserver.assertValuesOnly(3);
+    }
+
+    @Test
+    public void testDeliverLatestToView_ViewReady() throws Exception {
+        mPresenter.create();
+        mPresenter.attachView(mView);
+
+        TestObserver<Integer> testObserver = new TestObserver<>();
+        Observable.just(1, 2, 3)
+                .compose(RxTiPresenterUtils.<Integer>deliverLatestToView(mPresenter))
+                .subscribe(testObserver);
+
+        testObserver.assertNotComplete();
+        testObserver.assertNoErrors();
+        testObserver.assertValuesOnly(1, 2, 3);
+    }
+
+    @Test
+    public void testDeliverLatestToView_ViewNeverReady() throws Exception {
+        mPresenter.create();
+
+        TestObserver<Integer> testObserver = new TestObserver<>();
+        Observable.just(1, 2, 3)
+                .compose(RxTiPresenterUtils.<Integer>deliverLatestToView(mPresenter))
+                .subscribe(testObserver);
+
+        testObserver.assertNotComplete();
+        testObserver.assertNoErrors();
+        testObserver.assertEmpty();
+    }
+
+    @Test
+    public void testDeliverLatestToView_ViewComesAndGoes() throws Exception {
+        mPresenter.create();
+
+        PublishSubject<Integer> source = PublishSubject.create();
+        TestObserver<Integer> testObserver = new TestObserver<>();
+
+        source
+                .compose(RxTiPresenterUtils.<Integer>deliverLatestToView(mPresenter))
+                .subscribe(testObserver);
+
+        source.onNext(1);
+        source.onNext(2);
+        mPresenter.attachView(mView);
+        source.onNext(3);
+        mPresenter.detachView();
+        source.onNext(4);
+        source.onNext(5);
+        mPresenter.attachView(mView);
+        source.onNext(6);
+
+        testObserver.assertNotComplete();
+        testObserver.assertNoErrors();
+        testObserver.assertValuesOnly(2, 3, 5, 6);
+    }
+
+    @Test
+    public void testDeliverLatestToView_SingleItemViewComesAndGoes() throws Exception {
```

Comment:
We should add this test to rx1 as well. I'm actually not sure this is working.

Reply:
For that I've created a new issue. See #138